src/lib.rs

```diff
@@ -188,49 +188,54 @@ impl core::fmt::Display for PReg {
 #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
 #[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
 pub struct PRegSet {
-    bits: u128,
+    bits: [u128; 2],
 }
 
 impl PRegSet {
     /// Create an empty set.
     pub const fn empty() -> Self {
-        Self { bits: 0 }
+        Self { bits: [0; 2] }
     }
 
     /// Returns whether the given register is part of the set.
     pub fn contains(&self, reg: PReg) -> bool {
-        let bit = reg.index();
-        debug_assert!(bit < 128);
-        self.bits & 1u128 << bit != 0
+        debug_assert!(reg.index() < 256);
+        let bit = reg.index() & 127;
+        let index = reg.index() >> 7;
+        self.bits[index] & 1u128 << bit != 0
     }
 
     /// Add a physical register (PReg) to the set, returning the new value.
     pub const fn with(self, reg: PReg) -> Self {
-        let bit = reg.index();
-        debug_assert!(bit < 128);
-        Self {
-            bits: self.bits | (1u128 << bit),
-        }
+        debug_assert!(reg.index() < 256);
+        let bit = reg.index() & 127;
+        let index = reg.index() >> 7;
+        let mut out = self;
+        out.bits[index] |= 1u128 << bit;
+        out
```

Comment:
Now that this logic has become slightly more complex, perhaps we could delegate to `add` with something like

```
let mut out = self;
out.add(reg);
out
```
?

Reply:
I tried that at first, but it's not possible since `with` is a `const fn`. `add` can't be made `const` because it uses `&mut self`.